h2/src/main/org/h2/value/Transfer.java

```diff
@@ -533,12 +570,16 @@ public void writeValue(Value v) throws IOException {
         case Value.INTERVAL_HOUR:
         case Value.INTERVAL_MINUTE:
             if (version >= Constants.TCP_PROTOCOL_VERSION_18) {
-                writeInt(type);
                 ValueInterval interval = (ValueInterval) v;
-                writeBoolean(interval.isNegative());
+                int ordinal = type - Value.INTERVAL_YEAR;
+                if (interval.isNegative()) {
+                    ordinal = ~ordinal;
+                }
+                writeInt(INTERVAL);
+                writeByte((byte) ordinal);
```

Comment:
Yes, there are only two main interval data types in the standard: year-month intervals (`YEAR`, `MONTH`, and `YEAR TO MONTH`) and date-time intervals (all others), but we decided to use 13 value types for them because H2 can't handle additional parameters of data types well enough.

There were some changes to improve this situation after that. After some additional improvements we may want to merge these 13 types into one or two value types. So I don't see a big reason to allocate 13 separate types in network layer. Both PageStore and MVStore use only one type, the same format is now implemented in network layer.